nexus/db-model/src/zpool.rs

```diff
@@ -27,25 +27,28 @@ pub struct Zpool {
     // The physical disk to which this Zpool is attached.
     pub physical_disk_id: Uuid,
 
-    // TODO: In the future, we may expand this structure to include
-    // size, allocation, and health information.
-    pub total_size: ByteCount,
+    // There is much more information we can extract, including:
+    // - Size used, health, etc.
+    //
+    // However, we should also consider adding this information to "inv_zpool"
+    // instead, which acts more as the "observed" zpool.
+    pub total_size: Option<ByteCount>,
 }
 
 impl Zpool {
     pub fn new(
         id: Uuid,
         sled_id: Uuid,
         physical_disk_id: Uuid,
-        total_size: ByteCount,
+        _total_size: ByteCount,
```

Comment:
This callsite should be cleaned up, if we're really going the "get-size-from-inventory" route